src/torcontrol.cpp

```diff
@@ -522,9 +522,9 @@ void TorController::auth_cb(TorControlConnection& _conn, const TorControlReply&
     if (reply.code == 250) {
         LogPrint(BCLog::TOR, "tor: Authentication successful\n");
 
-        // Now that we know Tor is running setup the proxy for onion addresses
-        // if -onion isn't set to something else.
-        if (gArgs.GetArg("-onion", "") == "") {
+        // Now that we know Tor is running, setup the proxy for onion addresses
+        // if -onion and -proxy isn't set to something else.
+        if (gArgs.GetArg("-onion", "") == "" && gArgs.GetArg("-proxy", "") == "") {
```

Comment:
I guess this will work in most cases.

Though *ideally* we'd query Net's state directly here for "is there a proxy for onion already" instead of indirectly divising it from the arguments. This would really solve the problem of the current code here.

Reply:
> I guess this will work in most cases.
> 
> Though _ideally_ we'd query Net's state directly here for "is there a proxy for onion already" instead of indirectly divising it from the arguments. This would really solve the problem of the current code here.

Hi, @qubenix  

``` Diff

diff --git a/src/torcontrol.cpp b/src/torcontrol.cpp
index bec64fbf2..b632513c3 100644
--- a/src/torcontrol.cpp
+++ b/src/torcontrol.cpp
@@ -524,7 +524,8 @@ void TorController::auth_cb(TorControlConnection& _conn, const TorControlReply&
 
         // Now that we know Tor is running, setup the proxy for onion addresses
         // if -onion and -proxy isn't set to something else.
-        if (gArgs.GetArg("-onion", "") == "" && gArgs.GetArg("-proxy", "") == "") {
+        proxyType addrOnion;
+        if (gArgs.GetArg("-onion", "") == "" && !GetProxy(NET_ONION, addrOnion)) {
             CService resolved(LookupNumeric("127.0.0.1", 9050));
             proxyType addrOnion = proxyType(resolved, true);
             SetProxy(NET_ONION, addrOnion);

```
This litle patch should do the trick and fix the problem and that should be in line what @laanwj suggested, please feel free to copy paste. ;-)